### PR TITLE
Stops Ghosts changing lathe output

### LINKED ISSUE #82144: Ghosts can change where autolathes/techfabs drop printed items
## Issue Summary

https://github.com/tgstation/tgstation/assets/53777086/c2aa689a-a882-4cc5-b6ff-915da1e73dcf

### PR DESCRIPTION
**Pull request by Bilbo367**
## About The Pull Request

Fixes ghosts being able to change a lathes output direction.

## Why It's Good For The Game

Closes https://github.com/tgstation/tgstation/issues/82144 

## Changelog

:cl:
fix: Fixes ghosts being able to change a lathes output direction.
/:cl:

### CODE REVIEW

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -365,6 +365,8 @@
 
 /obj/machinery/autolathe/MouseDrop(atom/over, src_location, over_location, src_control, over_control, params)
 	. = ..()
+	if(isobserver(usr))
+		return
 	if((!HAS_SILICON_ACCESS(usr) && !isAdminGhostAI(usr)) && !Adjacent(usr))
```
Comment (SyncIt21) [inline code/game/machinery/autolathe.dm:370]:
```suggestion
	if(isobserver(usr) || (!HAS_SILICON_ACCESS(usr) && !isAdminGhostAI(usr)) && !Adjacent(usr))
```

Comment (SyncIt21) [reply code/game/machinery/autolathe.dm:370]: Write same checks for techfabs as well. They also have this functionality

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -365,7 +365,7 @@
 
 /obj/machinery/autolathe/MouseDrop(atom/over, src_location, over_location, src_control, over_control, params)
 	. = ..()
-	if((!HAS_SILICON_ACCESS(usr) && !isAdminGhostAI(usr)) && !Adjacent(usr))
+	if(isobserver(usr) || (!HAS_SILICON_ACCESS(usr) && !isAdminGhostAI(usr)) && !Adjacent(usr))
```
Comment (MrMelbert) [inline code/game/machinery/autolathe.dm:368]: If this works for observers, then it probably works for people who are stunned, crit, or dead. It should probably be performing some `can_interact` or `can_perform_action` check`.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/machinery/autolathe.dm (hunk 2 above)
Comment (Bilbo367) [reply code/game/machinery/autolathe.dm:368]: Trying out can_interact does not work. Not sure why i dont have AI_Interact enabled .

**Comment by san7890** [comment]
can you update the title of your PR to be more helpful? something like "fixes ghosts ability to mess with lathes" or something like that would be far more useful than what it is currently. thank you

**Comment by Bilbo367** [comment]
> can you update the title of your PR to be more helpful? something like "fixes ghosts ability to mess with lathes" or something like that would be far more useful than what it is currently. thank you

LOL i didnt even see that and fixed it and saw this. Yeah i need to be a little more careful with my titles, because i dont name my branches good.

**Comment by san7890** [comment]
> LOL i didnt even see that and fixed it and saw this. Yeah i need to be a little more careful with my titles, because i dont name my branches good.

no worries i do the same thing often (typically when github shits up my PR and deletes my title and submits a blank bodied PR)

**Pull request merged**